crawl-ref/source/attack.cc

```diff
@@ -1308,13 +1303,11 @@ int attack::calc_damage()
         potential_damage = using_weapon() || wpn_skill == SK_THROWING
             ? weapon_damage() : calc_base_unarmed_damage();
 
+        potential_damage = player_apply_weapon_skill(potential_damage, wpn_skill);
         potential_damage = player_stat_modify_damage(potential_damage);
+        potential_damage = player_apply_fighting_skill(potential_damage, false);
 
         damage = random2(potential_damage+1);
-
```

Comment:
I would prefer this now to be `damage = random2avg(potential_damage + 1, 3)` to make the chances of extreme low/high rolls less—specifically 3 because you removed two random2() rolls earlier on in the calculation.

Reply:
The old system multiplied 3 random2 rolls, which increases variance. Using random2avg would add 3 rolls instead, which reduces variance.

Reply:
Agree, sorry, I misread it :D